gcovr/tests/test_gcov_parser.py

```diff
@@ -276,3 +280,39 @@ def test_pathologic_codeline(capsys):
         'IndexError',
         '(ERROR) Exiting',
         'run gcovr with --gcov-ignore-parse-errors')
+
+
+def check_and_raise(number, total, exc_raised, queue_full):
+    queue_full.wait()
+    if number == 0:
+        raise Exception("Number == 0")
+    exc_raised.wait()
+    total[0] += 1
```

Comment:
Increments are not threadsafe. In this case it's not strictly important as we are only asserting an upper bound on this count. However, it might be better to append an item to this array here, and use the length of the array later.

Reply:
That is true - but it could mean that we miss failed tests if something goes wrong in the future?

Reply:
I have a preference for correct code because it prevents “WTF!!?!”s in the future. In this case, either a fix or a comment would be fine.